c++/src/kj/string.h

```diff
@@ -796,6 +800,19 @@ inline constexpr const char* ConstString::cStr() const { return content == nullp
 inline constexpr size_t String::size() const { return content == nullptr ? 0 : content.size() - 1; }
 inline constexpr size_t ConstString::size() const { return content == nullptr ? 0 : content.size() - 1; }
 
+inline constexpr void String::shrink(size_t size) {
+  if (content != nullptr) {
```

Comment:
This seems to be skipping the IREQUIRE in the case that the string is empty, but the check still seems relevant in that case.